core/src/credential.rs

```diff
@@ -0,0 +1,97 @@
+
+#![allow(unused_variables)]
+#![allow(dead_code)]
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+type VCContext = [String];
+
+pub(crate) const JSON_LD_CONTEXT:  VCContext = ["https://www.w3.org/2018/credentials/v1", "https://www.w3.org/2018/credentials/examples/v1"];
+pub(crate) const CRED_TYPE_PERMANENT_RESIDENT_CARD: str = "PermanentResidentCard";
+pub(crate) const CRED_TYPE_BANK_CARD: str = "BankCard";
+
+type CredentialSubject = serde_json::Value;
+
+enum CredType {
+    PermanentResidentCard(String),
+    BankCard(String)
+}
+
+pub trait VCCredential {
+    fn getCredType (&self) -> String;
+    fn getSubject (&self) -> String;
+    fn getContext (&self) -> [String];
+}
+
+impl VCCredential for VerifiableCredential<'_> {
+    fn getCredType (&self) -> String {
+        self.cred_type = ""
+    }
+    fn getSubject (&self) -> String {
+        self.subject = ""
+    }
+    fn getContext (&self) ->[String] {
+        return crate::credential::JSON_LD_CONTEXT
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct VerifiableCredential <'a> {
+    #[serde(rename = "@context")]
+	context:  VCContext,
+	id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cred_type:  Option<CredType>,
+	issuer: String,
+	issuance_date: String,
+	subject: CredentialSubject,
+	proof: *mut CredentialProof,
+}
+
+impl <'a> VerifiableCredential <'a>  {
+    pub fn init(cred_type: String, cred_subject: serde_json::Value, issuer: &'a str, id: &'a str) -> Self  {
```

Comment:
I think `new` might be more standard in Rust than `init` but I'm unsure

Reply:
I agree that `new` is more standard in terms of creating new struct, but motivation here is to underline that there is some object initialization logic, it is more like a factory then a constructor. Anyways, i'm happy to change according to language conventions. Maybe @Ryanmtate team has some PR review checklist that could help us in terms of language conventions and self-review?